r/R/type.R

```diff
@@ -187,8 +187,11 @@ NestedType <- R6Class("NestedType", inherit = DataType)
 #' @param timezone For `timestamp()`, an optional time zone string.
 #' @param byte_width byte width for `FixedSizeBinary` type.
 #' @param list_size list size for `FixedSizeList` type.
-#' @param precision For `decimal()`, precision
-#' @param scale For `decimal()`, scale
+#' @param precision For `decimal()`, the number of significant digits
+#'    the arrow `decimal` type can represent. Currently `decimal()` is mapped
+#'    to `DecimalType128`, having a maximum precision of 38 significant digits.
```

Comment:
Is decimal256 exposed? Should it be? If it is / if we do, should we mention what the max precision is for it?

Reply:
decimal256 is not exposed yet. I am planning on opening a bunch of tickets to:
* deprecate `decimal()` and replace it with `decimal128()`
* implement `decimal256()`. 
* maybe open another ticket to agree on a formalised approach to deprecation (both for functions and arguments). It would be relevant both for this PR and for `write_csv_arrow()`

Reply:
I added a comment on the ticket - we could record thoughts on implementing `decimal128()` and `decimal256()` there.